src/rust/core/server/src/threads/admin.rs

```diff
@@ -329,6 +353,147 @@ impl Admin {
+
+        let mut data = Vec::new();
+
+        for metric in &metrics::common::metrics::metrics() {
+            let any = match metric.as_any() {
+                Some(any) => any,
+                None => {
+                    continue;
+                }
+            };
+
+            if let Some(counter) = any.downcast_ref::<Counter>() {
+                data.push(format!("\"{}\": {}", metric.name(), counter.value()));
+            } else if let Some(gauge) = any.downcast_ref::<Gauge>() {
+                data.push(format!("\"{}\": {}", metric.name(), gauge.value()));
+            } else if let Some(heatmap) = any.downcast_ref::<Heatmap>() {
+                for (label, value) in PERCENTILES {
+                    let percentile = heatmap.percentile(*value).unwrap_or(0);
+                    data.push(format!("\"{}_{}\": {}", metric.name(), label, percentile));
+                }
+            }
+        }
+
+        data.sort();
+        let body = data.join(",");
+        let mut content = head;
+        content += &body;
+        content += "}";
+        content
+    }
+
+    fn prometheus_stats(&self) -> String {
+        let mut data = Vec::new();
+
+        for metric in &metrics::common::metrics::metrics() {
+            let any = match metric.as_any() {
+                Some(any) => any,
+                None => {
+                    continue;
+                }
+            };
+
+            if let Some(counter) = any.downcast_ref::<Counter>() {
+                data.push(format!(
+                    "# TYPE {} counter\n{} {}",
+                    metric.name(),
+                    metric.name(),
+                    counter.value()
+                ));
+            } else if let Some(gauge) = any.downcast_ref::<Gauge>() {
+                data.push(format!(
+                    "# TYPE {} gauge\n{} {}",
+                    metric.name(),
+                    metric.name(),
+                    gauge.value()
+                ));
+            } else if let Some(heatmap) = any.downcast_ref::<Heatmap>() {
+                for (label, value) in PERCENTILES {
+                    let percentile = heatmap.percentile(*value).unwrap_or(0);
+                    data.push(format!(
+                        "# TYPE {} gauge\n{}{{percentile=\"{}\"}} {}",
+                        metric.name(),
+                        metric.name(),
+                        label,
+                        percentile
+                    ));
+                }
+            }
+        }
+        data.sort();
+        let mut content = data.join("\n");
+        content += "\n";
+        let parts: Vec<&str> = content.split('/').collect();
+        parts.join("_")
+    }
+
+    /// Handle a HTTP request
+    fn handle_http_request(&self, request: Request) {
+        let url = request.url();
+        let parts: Vec<&str> = url.split('?').collect();
+        let url = parts[0];
+        match url {
+            "/metrics" => match request.method() {
```

Comment:
general comments about endpoint conventions (e.g. wrt prometheus collector) would be helpful